kasa/cli/main.py

```diff
@@ -102,186 +57,37 @@ def error(msg: str):
-        "--child-index",
-        "--index",
-        required=False,
-        default=None,
-        type=click.INT,
-        help=child_index_help,
-    )
-    async def wrapper(ctx: click.Context, dev, *args, child, child_index, **kwargs):
-        if child := await _get_child_device(dev, child, child_index, ctx.info_name):
-            ctx.obj = ctx.with_resource(patched_device_update(dev, child))
-            dev = child
-        return await ctx.invoke(wrapped_function, dev, *args, **kwargs)
-
-    # Update wrapper function to look like wrapped function
-    return update_wrapper(wrapper, wrapped_function)
-
-
-async def _get_child_device(
-    device: Device, child_option, child_index_option, info_command
-) -> Device | None:
-    def _list_children():
-        return "\n".join(
-            [
-                f"{idx}: {child.device_id} ({child.alias})"
-                for idx, child in enumerate(device.children)
-            ]
-        )
-
-    if child_option is None and child_index_option is None:
-        return None
-
-    if info_command in SKIP_UPDATE_COMMANDS:
-        # The device hasn't had update called (e.g. for cmd_command)
-        # The way child devices are accessed requires a ChildDevice to
-        # wrap the communications. Doing this properly would require creating
-        # a common interfaces for both IOT and SMART child devices.
-        # As a stop-gap solution, we perform an update instead.
-        await device.update()
-
-    if not device.children:
-        error(f"Device: {device.host} does not have children")
-
-    if child_option is not None and child_index_option is not None:
-        raise click.BadOptionUsage(
-            "child", "Use either --child or --child-index, not both."
-        )
-
-    if child_option is not None:
-        if child_option is OPTIONAL_VALUE_FLAG:
-            msg = _list_children()
-            child_index_option = click.prompt(
-                f"\n{msg}\nEnter the index number of the child device",
-                type=click.IntRange(0, len(device.children) - 1),
-            )
-        elif child := device.get_child_device(child_option):
-            echo(f"Targeting child device {child.alias}")
-            return child
-        else:
-            error(
-                "No child device found with device_id or name: "
-                f"{child_option} children are:\n{_list_children()}"
-            )
-
-    if child_index_option + 1 > len(device.children) or child_index_option < 0:
-        error(
-            f"Invalid index {child_index_option}, "
-            f"device has {len(device.children)} children"
-        )
-    child_by_index = device.children[child_index_option]
-    echo(f"Targeting child device {child_by_index.alias}")
-    return child_by_index
-
 
 @click.group(
     invoke_without_command=True,
-    cls=CatchAllExceptions(click.Group),
+    cls=CatchAllExceptions(LazyGroup),
+    lazy_subcommands={
+        "discover": "kasa.cli.discover.discover",
```

Comment:
It would be nicer if we wouldn't need to have a static list but just add the commands here?

For grouped commands it should be really straightforward, i.e., just by adding the group to the main one, but individual commands would need to be imported and added manually.

Reply:
The intention here is to avoid importing the commands up front and do it lazily to reduce the imports the cli does at startup: https://click.palletsprojects.com/en/8.1.x/complex/#lazily-loading-subcommands. I might not be following what you're suggesting though.

Reply:
Ok so I simplified this such that each submodule contains a group with the same name so only need to add the submodule name to the list.  For the existing subcommands under `device` and `light` these are still made available at the parent cli command but we won't need to keep adding.

Reply:
I think our imports do not (or at least they should not) perform anything performance-intensive operations, so having them directly referenced will avoid potential issues if something gets renamed etc.